fix: clean up type handling (#248)
GalSim-developers/JAX-GalSim: docs/sharp-bits.rst

   # JAX-GalSim — real_part is a copy
   real_part = complex_image.real  # independent array

Scalar Types, Array Types, and Casting
--------------------------------------

With the use of JAX, there are now many possible types for numeric data. These include

> - **Python scalars**: Things with types that are ``float``, ``int``, or ``complex``.

Reviewer: Consider 'Things' -> 'Objects' in this bullet point list?

Author: done!

> - **NumPy scalars**: Things with types that are subclasses are ``np.floating`` and ``np.integer``.

Reviewer: ```suggestion
- **NumPy scalars**: Things with types that are subclasses of ``np.floating`` and ``np.integer``.
```

- **Numpy array scalars**: Things with a type that is ``np.ndarray`` and has ``np.ndim(...) == 0``.

> - **Numpy arrays**: Things with a type that is ``np.ndarray`` and has ``np.ndim(...) > 0``.

Author: ```suggestion
- **NumPy array scalars**: Things with a type that is ``np.ndarray`` and has ``np.ndim(...) == 0``.
- **NumPy arrays**: Things with a type that is ``np.ndarray`` and has ``np.ndim(...) > 0``.
```

- **JAX array scalars**: Things with a type that is ``jnp.ndarray`` and has ``np.ndim(...) == 0``.

> - **JAX arrays**: Things with a type that is ``np.ndarray`` and has ``np.ndim(...) > 0``.

Author: ```suggestion
- **JAX array scalars**: Things with a type that is ``jax.numpy.ndarray`` and has ``jax.numpy.ndim(...) == 0``.
- **JAX arrays**: Things with a type that is ``jax.numpy.ndarray`` and has ``jax.numpy.ndim(...) > 0``.
```

> **JAX does not have pure scalar types like NumPY. JAX uses array scalars for those instead.**

Reviewer: ```suggestion
**JAX does not have pure scalar types like NumPy. JAX uses array scalars for those instead.**
```

should not affect scientific conclusions.

⚠️ Additional Sharp Bits
------------------------

In the :doc:`api/index` you will find **🔪 JAX-GalSim - The Sharp Bits 🔪** blocks highlighting additional important caveats for specific classes and or methods. These could include things like:

> - Some classes do not perform some of Galsim's test for correctness during initialization (e.g., :meth:`~jax_galsim.GSObject.drawImage`).

Reviewer: I just noticed this comment refers to a class but the example is a method, perhaps we can change it to InterpolatedImage instead where I believe there are still some checks during initialization that we don't do.

```suggestion
- Some classes do not perform some of Galsim's test for correctness during initialization (e.g., :meth:`~jax_galsim.InterpolatedImage`).
```
